src/TermAuctionList.sol

```diff
@@ -186,9 +186,18 @@ library TermAuctionList {
                 // because the offerLocker will have already removed the offer. 
                 // This applies if the repoToken hasn't been added to the repoTokenList 
                 // (only for new auctions, not reopenings).  
-                repoTokenListData.validateAndInsertRepoToken(
+                (bool isValidRepoToken, , uint256 redemptionTimestamp ) = repoTokenListData.validateAndInsertRepoToken(
                     ITermRepoToken(offer.repoToken), discountRateAdapter, asset
                 );
+                if (!isValidRepoToken && block.timestamp > redemptionTimestamp) {

```

Comment:
If `!isValidRepoToken` because the discount rate is 0, then you don't want to redeem the tokens?

Reply:
With the try/catch I believe that this statement should be enough for redemptions that we would want to run. Of course if the discount rate is 0 and the redemptionTimestamp hasn't passed, we can't redeem (redemptions are closed before the redemption timestamp).

Reply:
But, if:
- Auction is completed
- Discount rate is 0 
- Redemption timestamp hasn't passed
Then, the offer will be removed from the list without the repoToken being added to the repoTokenList, which means it will never be redeemed.

Reply:
Got it, will remove the zero discount rate check in that case.